### ISSUE TITLE
Support for bibtex citekeys

### ISSUE THREAD

**Issue by Cutuchiqueno** [anoopkcn/obsidian-reference-map#7]
This is an amazing plugin for which I looked out for for a long time. However, most of my references are pandoc citekeys referring to a citation database (Zotero, bibtex,...). I think it is a typical situation, especially in academia. So, without being able to include those, 80% of my references would not be included and the value of the plug-in diminishes. Would it be very complicated to add and parse citation keys? Do you know the reference map that Zettlr creates in this fashion: https://docs.zettlr.com/en/academic/citations/ and https://docs.zettlr.com/en/core/sidebar/

**Comment by anoopkcn** [comment]
Hi @Cutuchiqueno , It's a very nice idea.  I am trying to avoid the Pandoc dependency for this plugin, but I haven't found a solution to this yet.  One idea is to let the user define a citation regex, and then I can find the citation keys in the CSL JSON file and get the reference ID. But this method is very error-prone since the user might not always give the regex pattern correctly or forgets to update the format after changing it in Zotero. So it is a very inelegant solution.  I will think about it a bit more, but If nothing works, I will implement the Pandoc version like Zettlr did.

**Comment by Cutuchiqueno** [comment]
That's great. If you need support in some way, as far as capacities allow, just tell me.

>  I don't know if this is your use case

Well, I would say, my individual use-case is not the important one, but what is probable useful for most people. Here, I still think, pandoc citation format is one of the most common approaches. Think of markdown based academic writing frameworks such as knitr and alike that also use pnadoc citation format for references in texts (as well as CSL-YML as source format; https://bookdown.org/yihui/rmarkdown-cookbook/bibliography.html).

In the Obsidian ecosystem the Zoter Integration extension (https://github.com/mgmeyers/obsidian-zotero-integration), offers a feature to just start writing `@...`  (pandoc citation format) and it triggers suggestions from the citekeys within your Zotero database. Thus, choosing this format could also mean to neatly integrate with other extensions, relevant within this area.

So basically, that would be my rational behind arguing in favour of the pandoc citation format. But of cause this does not mean, that there could be another reasoning; just wanted to offer my thoughts...

Pandoc citations: https://pandoc.org/MANUAL.html#citations-in-note-styles

**Comment by anoopkcn** [comment]
I implemented this feature in the latest version of the plugin. You can enable this in the settings tab. One has to provide the `Better BibTeX CSL JSON ` file from Zotero for this to work.

**Comment by Cutuchiqueno** [comment]
amazing, I will test immediately :D

**Comment by anoopkcn** [comment]
👍 You can use `Better CSL JSON` and turn on background updates in Zotero to keep the citations updated.

**Comment by anoopkcn** [comment]
Thanks for your kind words. No need for any donations. If you have any suggestions or find any bugs in the future, I'm happy to help. You can close the issue if there are no further comments about this.
